### DISCUSSION TITLE
IMXRT1170 secure boot using HAB

### DISCUSSION THREAD

**Discussion by danielb-tkjcy3rq** [zephyrproject-rtos/zephyr#79697] [Q&A] [answered]
Hi All,

I have a custom imxrt1170 board running running multi-core Zephyr (3.6.0) image and mcuboot (2.1.0). MCUBoot verifying the Zephyr signature before booting it is all working fine. The bit I'm not clear on is getting the IMXRT ROM code to verify the MCUboot image using the HAB. I've done this previously on imx6 solo and imx6ul devices with uboot. Is there support for this on Zephyr + MCUBoot?

Thanks

**Comment by DerekSnell** [comment, marked as answer]
Hi @danielb-tkjcy3rq ,
Sorry for the late reply, and I realize this response is probably too late for you to be of help.

NXP's [MCUXpresso Secure Provisioning Tool](https://www.nxp.com/design/design-center/software/development-software/mcuxpresso-software-and-tools-/mcuxpresso-secure-provisioning-tool:MCUXPRESSO-SECURE-PROVISIONING?tid=vanMCUXPRESSO-SECURE-PROVISIONING) provisions the HAB and ROM bootloader for secure boot.  You can build the MCUboot image with Zephyr, and then use this tool to sign that image and prepare it for the bootloader.  Best regards
